class.jetpack.php

```diff
@@ -3253,6 +3253,43 @@ public static function unlink_user( $user_id = null ) {
 		return true;
 	}
 
+	public static function update_disconnection_notice( $user_id ) {
+		$notice = (array) Jetpack_Options::get_option( 'disconnection_notice' );
+		$notice = array_merge( $notice, array( $user_id ) );
+		Jetpack_Options::update_option( 'disconnection_notice', $notice );
+	}
+
+	public static function unlink_user_on_unknown_token( $user_id ) {
+		$master_user = (int) Jetpack_Options::get_option( 'master_user' );
+		if ( $user_id == 0 || $user_id == JETPACK_MASTER_USER ) {
+			$user_id = $master_user;
+		}
+
+		// unset the user token.
+		$tokens = Jetpack_Options::get_option( 'user_tokens' );
+		unset( $tokens[$user_id] );
+		Jetpack_Options::update_option( 'user_tokens', $tokens );
+
+		if ( $user_id !== $master_user ) {
+			self::update_disconnection_notice( $user_id );
+			return;
+		}
+
+		require_once( JETPACK__PLUGIN_DIR .'sync/class.jetpack-sync-users.php' );
+		Jetpack_sync_users::maybe_demote_master_user( $master_user, true );
+		if ( $master_user !== Jetpack_Options::get_option( 'master_user' ) ) {
+			// Master user switched
+			self::update_disconnection_notice( $user_id );
+			return;
+		}
+
+		// Delete Everything - disconnect the site
```

Comment:
How is the wpcom-side of this being handled? If we just delete the tokens on the client site, won't wpcom still think it's connected and the site will run into issues when trying to reconnect?

Reply:
This is only intended to delete tokens that return an unknown_error. Which happens when the tokens are not found on the .com side any more.